### PR TITLE
Cw 462 monero polyseed restore support

### PR DESCRIPTION
**Pull request by konstantinullrich**
# Description

Add Polyseed to restore a Monero Wallet

Requires [Monero PR-3](https://github.com/cake-tech/monero/pull/3) and [Monero PR-4](https://github.com/cake-tech/monero/pull/4)

# Pull Request - Checklist  

- [x] Initial Manual Tests Passed
- [x] Double check modified code and verify it with the feature/task requirements
- [x] Format code
- [x] Look for code duplication
- [x] Clear naming for variables and methods

### CODE REVIEW

File: cw_monero/ios/Classes/monero_api.cpp [hunk 1]
```diff
@@ -438,6 +467,11 @@ extern "C"
 
     const char *seed()
     {
+        std::string _rawSeed = get_current_wallet()->getCacheAttribute("cakewallet.seed");
+        if (!_rawSeed.empty())
```
Comment (OmarHatem28) [inline cw_monero/ios/Classes/monero_api.cpp:471]: are we sure this won't return a null pointer if not available?

Comment (konstantinullrich) [reply cw_monero/ios/Classes/monero_api.cpp:471]: Yes, according to the Monero `wallet2_api.h` doc-strings 
```
/*!
* \brief getCacheAttribute - return an arbitrary string attached to a wallet cache attribute
* \param key - the key
* \return the attached string, or empty string if there is none
*/
```

File: cw_monero/lib/monero_wallet_service.dart [hunk 2]
```diff
@@ -234,6 +251,42 @@ class MoneroWalletService extends WalletService<
...
+      rethrow;
+    }
+  }
+
+  Future<MoneroWallet> _restoreFromPolyseed(String path, String password, Polyseed polyseed,
+      WalletInfo walletInfo, PolyseedLang lang,
+      {PolyseedCoin coin = PolyseedCoin.POLYSEED_MONERO}) async {
+    final height = getMoneroHeigthByDate(
+        date: DateTime.fromMillisecondsSinceEpoch(polyseed.birthday * 1000));
+    final spendKey = keyToHexString(polyseed.generateKey(coin, 32));
```
[the remark this answers is not in the document]
Comment (konstantinullrich) [reply cw_monero/lib/monero_wallet_service.dart:274]: The `32` refers to the size of the key generated here. In order to assure compatibility with e.g. feather wallet I took it from there.

File: lib/view_model/settings/sync_mode.dart [hunk 3]
```diff
@@ -11,5 +11,6 @@ class SyncMode {
     SyncMode("Disabled", SyncType.disabled, Duration.zero),
     SyncMode("Unobtrusive", SyncType.unobtrusive, Duration(days: 1)),
     SyncMode("Aggressive", SyncType.aggressive, Duration(hours: 6)),
+    SyncMode("Super Aggressive", SyncType.aggressive, Duration(minutes: 15)),
```
[the remark this answers is not in the document]
Comment (konstantinullrich) [reply lib/view_model/settings/sync_mode.dart:14]: Good Catch

**Review by OmarHatem28 — APPROVED** [review]

**Pull request merged**